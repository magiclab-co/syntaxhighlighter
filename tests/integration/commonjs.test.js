import sizzle from 'sizzle';
import SyntaxHighlighter, { registerBrush, clearRegisteredBrushes } from '../..';
import Brush from '../fixtures/test_brush_v4';

function setupSyntaxHighlighter(html) {
  let div;

  beforeAll(() => {
    div = document.createElement('div');
    div.innerHTML = html;
    document.body.appendChild(div);

    SyntaxHighlighter.highlight({ gutter: false });

    return new Promise((resolve) => {
      setTimeout(resolve, 900);
    });
  });

  afterAll(() => {
    document.body.removeChild(div);
  });
}

describe('integration/commonjs', () => {
  beforeAll(() => {
    registerBrush(Brush);
  });

  afterAll(() => {
    clearRegisteredBrushes();
  });

  describe('first render pass', () => {
    setupSyntaxHighlighter(`<pre class="brush: test_brush_v4;">first</pre>`);

    it('has applied the brush', () => {
      expect(sizzle('.syntaxhighlighter')[0].innerHTML).toMatchSnapshot();
    });

    it('does not render gutter', () =>
      expect(sizzle('.syntaxhighlighter td.gutter').length).toBe(0));
  });

  describe('second render pass', () => {
    setupSyntaxHighlighter(`<pre class="brush: test_brush_v4;">second</pre>`);

    it('has applied the brush', () => {
      expect(sizzle('.syntaxhighlighter')[0].innerHTML).toMatchSnapshot();
    });
  });
});
