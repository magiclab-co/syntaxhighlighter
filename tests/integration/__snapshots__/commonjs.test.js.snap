// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration/commonjs first render pass has applied the brush 1`] = `
"
        <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
          
          <tbody>
            <tr>
              
              <td class=\\"code\\">
                <div class=\\"container\\"><div class=\\"line number1 index0 alt2\\"><code class=\\"test_brush_v4 plain\\">first</code></div></div>
              </td>
            </tr>
          </tbody>
        </table>
      "
`;

exports[`integration/commonjs second render pass has applied the brush 1`] = `
"
        <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
          
          <tbody>
            <tr>
              
              <td class=\\"code\\">
                <div class=\\"container\\"><div class=\\"line number1 index0 alt2\\"><code class=\\"test_brush_v4 plain\\">second</code></div></div>
              </td>
            </tr>
          </tbody>
        </table>
      "
`;
